tests/libtest.sh

```diff
@@ -696,18 +696,20 @@ skip_without_fuse () {
     [ -e /etc/mtab ] || skip "no /etc/mtab"
 }
 
-has_gpgme () {
-    local ret
+has_ostree_feature () {
+    local ret=0
+    # Note that this needs to write to a file and then grep the file, to
+    # avoid ostree --version being killed with SIGPIPE and exiting with a
+    # nonzero status under `set -o pipefail`.
```

Comment:
Oh man...wow.

Reply:
As I said, more involved than I expected. This PR brought to you by the dept. of Unix arcana.

Reply:
Even though it's eminently useful, I stopped using `pipefail` years ago after being burned by the same issue in `tar`. It might be nice to install a `SIGPIPE` handler in the [version printing code](https://github.com/ostreedev/ostree/blob/1b3ffe54e87c108b576a528efa6493f935a80f27/src/ostree/ot-main.c#L511) since this is likely to affect other `ostree` consumers.

Reply:
One thing I've said many times with coworkers in live video chats, but rarely externally is that a nontrivial component of why I still (20+ years later) wake up and log into my computer for this job is that I just really enjoy collaborating with lots of smart people across the internet.  Also there's the continual "surprise" factor in computers as in "what did someone figure out how to make them do this year" (and right now it's AI).

Anyways this little bit combines those two - I just want to say thanks @smcv as one of those smart people on the Internet for everything you do :pray: - and this little bit of pipefail is one of those surprises :smile:

Reply:
See #3203.